2011-04-01  Aurelien Jarno  <aurelien@aurel32.net>

        * sysdeps/unix/sysv/linux/alpha/sysconf.c: Fix cache sysconf
	switch.

---
 ports/sysdeps/unix/sysv/linux/alpha/sysconf.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/ports/sysdeps/unix/sysv/linux/alpha/sysconf.c
+++ b/ports/sysdeps/unix/sysv/linux/alpha/sysconf.c
@@ -135,7 +135,7 @@
   if (shape <= 0)
     return shape;
 
-  switch (name % 3)
+  switch ((name - _SC_LEVEL1_ICACHE_SIZE) % 3)
     {
     case 0: /* total size */
       return shape & -0x100;
